File: Competitive Programming 3/Section 1.4/Real Life Easier/362 - 18,000 Seconds Remaining/Python/main.py
import math

def main():
    cs = 1 
    while True:
        remain = int(input())
        if not remain :
            return
        print('Output for data set {}, {} bytes:'.format(cs,remain))
        i , j  = 0 , 0
        acum5 = 0 
        while remain:
            x = int(input())
            remain-=x
            acum5 += x
            if i == 4:
                #process acum5
                remain5 = remain * 5
                print('   Time remaining: {}'.format('{} seconds'.format(remain5//acum5 + (1 if remain5%acum5 else 0)) if acum5 else 'stalled'))
                acum5=0 
            i=(i+1)%5
            j+=1
        print('Total time: {} seconds\n'.format(j))
        cs+=1
main()

File: Competitive Programming 3/Section 1.4/Real Life Easier/362 - 18,000 Seconds Remaining/C++/main.cpp
#include<bits/stdc++.h>

using namespace std;

int main() {
    ios_base::sync_with_stdio(0);cin.tie(0);
    int remain,x;
    for(int cs = 1 ;cin >>remain && remain;cs++){
        cout << "Output for data set " << cs << ", " << remain << " bytes:\n";
        int j=0;
        for(int i=0,acum5=0;remain;i=(i+1)%5,j++){
            cin>>x;
            remain-=x;
            acum5+=x;
            if(i==4){
                //process acum5
                auto remain5 = remain*5 ;
                cout << "   Time remaining: " << (acum5 ? (to_string(remain5/acum5+(remain5%acum5?1:0))+" seconds"):"stalled") << "\n";
                acum5=0;
            }
        }
        cout << "Total time: " << j << " seconds\n\n";
    }
    return 0;
}